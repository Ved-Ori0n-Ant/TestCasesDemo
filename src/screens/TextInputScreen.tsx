import React, {useState} from 'react';
import {SafeAreaView, StyleSheet, Text, TextInput, TouchableOpacity, View} from 'react-native';
import NextnPrevious, {Stack} from '../components/NextnPrevious';

const TextInputScreen = () => {
  const [text, setText] = useState('');
  return (
    <SafeAreaView style={styles.body}>
      <View style={styles.subBody}>
        <TextInput
          placeholder={'Name'}
          testID={'TextInput'}
          value={text}
          onChangeText={(txt: string) => setText(txt)}
          style={{
            borderRadius: 12,
            borderWidth: 2,
            borderColor: '#000',
            padding: 16,
            fontSize: 28,
            marginHorizontal: 8,
          }}
        />
      </View>
      <View style={styles.text}>
      <Text testID='text'>{`Name is: ${text}`}</Text>
      </View>
      <NextnPrevious currentScreen={Stack.TextInputScreen} />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  body: {
    flex: 1,
  },
  subBody: {
    flex: 1,
    justifyContent: 'center',
  },
  text: {
    marginBottom: 250,
    justifyContent: 'center',
    alignItems: 'center'
  }
});

export default TextInputScreen;
