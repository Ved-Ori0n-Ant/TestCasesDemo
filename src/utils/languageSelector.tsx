import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Dimensions,
  Image,
} from 'react-native';
import {useTranslation} from 'react-i18next';
import '../localization';

const {width} = Dimensions.get('window');

const LANGS = [
  {lngCode: 'en', label: 'English'},
  {lngCode: 'hi', label: 'हिन्दी'},
  {lngCode: 'ar', label: 'عربي'},
];

const LanguageSelector = () => {
  const {t, i18n} = useTranslation();
  const selectedLngCode: any = i18n.language;
  const setLng = (lngCode: string) => i18n.changeLanguage(lngCode);
  return (
    <View style={styles.container}>
      <View style={styles.titleContainer}>
        {/* Should return select your language in respective language */}
        <Text style={styles.select}>{t('languageSelector:selectLng')}</Text> 
      </View>
      {LANGS.map((l) => {
        const selected = l.lngCode === selectedLngCode;
        return (
          <TouchableOpacity
            onPress={() => setLng(l.lngCode)}
            key={l.lngCode}
            disabled={selected}>
            <View style={[styles.row, selected ? styles.selectedRow : {}]}>
              <Text style={[selected ? styles.selectedText : styles.text]}>
                {l.label}
              </Text>
              {selected && <Image source={require('../utils/images/world.png')} style={{height: 20, width:20}} />}
            </View>
          </TouchableOpacity>
        );
      })}
    </View>
  );
};

export default LanguageSelector;

const styles = StyleSheet.create({
  container: {
    padding: 10,
    backgroundColor: 'rgb(231, 232, 235)',
    width: width * 0.8,
  },
  titleContainer: {
    alignItems: 'center',
    paddingBottom: 20,
  },
  select: {
    fontSize: 20,
  },
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 10,
    paddingVertical: 5,
  },
  selectedRow: {
    backgroundColor: 'rgb(45, 45, 45)',
  },
  selectedText: {
    color: 'rgb(231, 232, 235)',
  },
  text: {
    color: 'rgb(45, 45, 45)',
  },
});