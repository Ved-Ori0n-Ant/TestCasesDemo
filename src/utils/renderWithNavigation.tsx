import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {render} from '@testing-library/react-native';

export function renderWithNavigation(renderComponent: any) {
  return render(<NavigationContainer>{renderComponent}</NavigationContainer>);
}
