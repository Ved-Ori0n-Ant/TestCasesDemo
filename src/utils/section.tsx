import React, {ReactNode} from 'react';
import {StyleSheet, View, Text} from 'react-native';

interface Props {
  children: ReactNode;
  title: string;
  testID?: string;
}

const Section = ({title, children, testID}: Props) => {
  return (
    <View style={styles.container} testID={testID}>
      <View style={styles.titleContainer}>
        <Text style={styles.title}>{title}</Text>
      </View>
      {children}
    </View>
  );
};

export default Section;

const styles = StyleSheet.create({
  container: {
    alignContent: 'center',
    marginVertical: 10,
    padding: 10,
    justifyContent: 'center',
    width: '80%',
  },
  titleContainer: {
    alignItems: 'center',
  },
  title: {
    fontSize: 26,
    fontWeight: 'bold',
    color: 'maroon',
  },
});