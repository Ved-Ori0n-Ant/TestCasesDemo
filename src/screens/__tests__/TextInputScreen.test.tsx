import 'react-native';
import React from 'react';
import { renderWithNavigation } from '../../utils/renderWithNavigation';
import TextInputScreen from '../TextInputScreen';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';

describe('Tests for TextInputScreen', () => {
    it('should render overall', ()=>{
        const tree = renderWithNavigation(<TextInputScreen />)
        expect(tree.toJSON).toMatchSnapshot();
    });
    it('should get textInput field', () => {
        const tree = renderWithNavigation(<TextInputScreen />)
        expect(tree.getByPlaceholderText('Name')).toBeDefined();
        expect(tree.getByTestId('TextInput')).toBeDefined();
    });
    it('should enter the name', () => {
        const tree = renderWithNavigation(<TextInputScreen />)
        const inputString = tree.getByTestId('TextInput');
        fireEvent.changeText(inputString, 'asdf');
        // console.log('!!', tree.getByTestId('TextInput').props)
        expect(inputString.props.value).toBe('asdf');
    });
    it('should update the text with it', ()=>{
        const tree = render(<NavigationContainer><TextInputScreen /></NavigationContainer>);
        const inputString = tree.getByTestId('TextInput');
        const text = tree.getByTestId('text');
        fireEvent.changeText(inputString, 'asdf');
        expect(text.props.children).toBe('Name is: asdf')
    });
});
