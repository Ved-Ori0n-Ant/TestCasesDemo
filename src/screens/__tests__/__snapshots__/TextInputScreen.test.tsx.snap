// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for TextInputScreen should render overall 1`] = `[Function]`;
